integration_test/cases/preload.exs

```diff
@@ -552,6 +552,30 @@ defmodule Ecto.Integration.PreloadTest do
     assert p2.id == c4.post.id
   end
 
+  test "custom preload_order" do
+    p1 = TestRepo.insert!(%Post{title: "1"})
+
+    %Comment{id: cid1} = TestRepo.insert!(%Comment{text: "2", post_id: p1.id})
+    %Comment{id: cid2} = TestRepo.insert!(%Comment{text: "1", post_id: p1.id})
+
+    %User{id: uid1} = TestRepo.insert!(%User{name: "bar"})
+    %User{id: uid2} = TestRepo.insert!(%User{name: "foo"})
+
+    TestRepo.insert_all("posts_users", [[post_id: p1.id, user_id: uid1], [post_id: p1.id, user_id: uid2]])
```

Comment:
Please correct me if I'm wrong (writing from memory), but we can write it much shorter:
```suggestion
    TestRepo.insert!(%PostUser{post_id: p1.id, user: %{name: "bar"}})
    TestRepo.insert!(%PostUser{post_id: p1.id, user: %{name: "foo"}})
```
or even:
```suggestion
    p1 = TestRepo.insert!(%Post{users: [%{name: "bar"}, %{name: "foo"}], title: "1"})
```

Reply:
@Eiji7 Updated to use the second option. Thanks for the suggestion.